packages/firebase_analytics/firebase_analytics/lib/firebase_analytics.dart

```diff
@@ -13,6 +13,9 @@ const MethodChannel firebaseChannel =
 
 /// Firebase Analytics API.
 class FirebaseAnalytics {
+  FirebaseAnalytics._();
+  static final instance = FirebaseAnalytics._();
```

Comment:
Consider using a [factory to return the singleton instance](https://stackoverflow.com/questions/12649573/how-do-you-build-a-singleton-in-dart), and thus not break current code.

Reply:
The intent isn't to make the default constructor return a singleton, but to match the semantics of other firebase packages by providing a static instance getter.

Reply:
No, I’d say I’m the one missing context. I was trying to avoid breaking APIs, but consistency is also important.